src/vm/threadpoolrequest.h

```diff
@@ -159,22 +161,31 @@ class ManagedPerAppDomainTPCount : public IPerAppDomainTPCount {
     inline void SetAppDomainUnloading()
     {
         LIMITED_METHOD_CONTRACT;
-        m_numRequestsPending = ADUnloading;
+        VolatileStore(&m_numRequestsPending, ADUnloading);
     }
 
     inline void ClearAppDomainUnloading();
 
     inline BOOL IsAppDomainUnloading()
     {
-        return m_numRequestsPending.Load() == ADUnloading;
+        return VolatileLoad(&m_numRequestsPending) == ADUnloading;
     }
 
     void DispatchWorkItem(bool* foundWork, bool* wasNotRecalled);
 
 private:
-    Volatile<LONG> m_numRequestsPending;
-    ADID m_id;
-    TPIndex m_index;
+    struct {
```

Comment:
K, I was struggling getting some of the alignments to stick; but I think `DECLSPEC_ALIGN(64)` fixed most of them.